tools/agent_tool/src/cpp/user_interface/arguments_configuration.h

```diff
@@ -0,0 +1,206 @@
+struct Arg : public option::Arg
+{
+    static void print_error(
+            const char* msg1,
+            const option::Option& opt,
+            const char* msg2)
+    {
+        fprintf(stderr, "%s", msg1);
+        fwrite(opt.name, opt.namelen, 1, stderr);
+        fprintf(stderr, "%s", msg2);
+    }
+
+    static option::ArgStatus Unknown(
+            const option::Option& option,
+            bool msg)
+    {
+        if (msg)
+        {
+            print_error("Unknown option '", option, "'\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus Required(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0 && option.arg[0] != 0)
+        {
+            return option::ARG_OK;
+        }
+
+        if (msg)
+        {
+            print_error("Option '", option, "' requires an argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus Numeric(
+            const option::Option& option,
+            bool msg)
+    {
+        char* endptr = 0;
+        if (option.arg != 0 && strtol(option.arg, &endptr, 10))
+        {
+        }
+        if (endptr != option.arg && *endptr == 0)
+        {
+            return option::ARG_OK;
+        }
+
+        if (msg)
+        {
+            print_error("Option '", option, "' requires a numeric argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus String(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0)
+        {
+            return option::ARG_OK;
+        }
+        if (msg)
+        {
+            print_error("Option '", option, "' requires a string argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus EntityKind(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0)
+        {
+            std::string data_type = std::string(option.arg);
+            if (data_type != CLIENT_ENTITY_KIND_ARG &&
+                    data_type != SERVER_ENTITY_KIND_ARG &&
+                    data_type != REPEATER_ENTITY_KIND_ARG)
+            {
+                if (msg)
+                {
+                    print_error("Option '", option, "' only accepts <client|server|repeater> values\n");
+                }
+                return option::ARG_ILLEGAL;
+            }
+            return option::ARG_OK;
+        }
+        if (msg)
+        {
+            print_error("Option '", option, "' requires a string argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+    static option::ArgStatus Transport(
+            const option::Option& option,
+            bool msg)
+    {
+        if (option.arg != 0)
+        {
+            std::string transport = std::string(option.arg);
+            if (transport != "shm" && transport != "udp" && transport != "udpv4" && transport != "udpv6")
```

Comment:
It should accept minimum udpv4 and tcpv4. Rest of transports are not relevant so far